src/main/java/ua/tc/marketplace/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package ua.tc.marketplace.service.impl;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import ua.tc.marketplace.config.UserDetailsImpl;
+import ua.tc.marketplace.model.entity.User;
+import ua.tc.marketplace.repository.UserRepository;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    @Autowired
+    private UserRepository repository;
```

Comment:
I suggest to use constructor injection here (@RequiredArgsConstructor)

Reply:
added